File: src/types/reservation.ts
import type { Room } from './room';
import type { Customer } from './customer';

export interface Reservation {
  id: number;
  customerId: number;
  roomId: number;
  checkInDate: string;
  checkOutDate: string;
  totalPrice: number;
  room?: Room;
  customer?: Customer;
}

export interface ReservationRequest {
  customerId: number;
  roomId: number;
  checkInDate: string;
  checkOutDate: string;
}

File: otel-api/Models/Reservation.cs
namespace otel_api.Models
{
    public class Reservation
    {
        public int Id { get; set; }
        public int CustomerId { get; set; }
        public int RoomId { get; set; }
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public decimal TotalPrice { get; set; }

        // İlişkiler (Include için gerekli)
        public Room? Room { get; set; }
        public Customer? Customer { get; set; }
    }
}